### PR TITLE
feat: add edit TextResponse fields

### PR DESCRIPTION
**Pull request by storyworks**
# Description

Please include a summary of the change and which todo is completed. Please also include relevant motivation and context. List any dependencies that are required for this change.

[Field properties](https://3.basecamp.com/3105655/buckets/29284565/todos/5326812837)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Can change label
- [ ] Label cannot be empty
- [ ] Can change hint text
- [ ] Can change minRows 

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.stories.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const TextResponseStory = {
   ...journeysAdminConfig,
   component: TextResponseFields,
   title:
-    'Journeys-Admin/Editor/ControlPanel/Attributes/TextResponse/TextResponse'
+    'Journeys-Admin/Editor/ControlPanel/Attributes/TextResponse/TextResponseFields'
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.stories.tsx:13]: Nice find

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx [hunk 2]
```diff
@@ -1,13 +1,86 @@
...
+          minRows
+        }
+      }
+    })
+  }
+
+  const options = [
+    {
+      value: 1,
+      label: 'One Row'
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx:59]: Put in translation file

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          </EditorProvider>
+        </JourneyProvider>
+      </MockedProvider>
+    )
+    expect(getByRole('textbox', { name: 'Label' })).toBeInTheDocument()
+    expect(getByRole('textbox', { name: 'Hint' })).toBeInTheDocument()
+    expect(getByRole('button', { name: 'One Row' })).toBeInTheDocument()
+  })
+
+  it('should change rows of text response', async () => {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.spec.tsx:29]: Nice and also add test about how the default row size is 3 (after moving all the minRows related code to a new file like how `Label` and `Hint` is seperated)

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      value: 4,
+      label: 'Four Rows'
+    }
+  ]
+
+  return (
+    <Stack>
+      <Label />
+      <Hint />
+      <ToggleButtonGroup
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx:79]: I asked Steven about adding a label above this in Figma, awaiting his response - are you able to add / resolve this comment when you find out his response?

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Label/Label.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          __typename: 'TextResponseBlock',
+          label: target.value
+        }
+      }
+    })
+  }
+
+  const initialValues =
+    selectedBlock != null ? { textResponseLabel: selectedBlock.label } : null
+  const maxCharacters = 16
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Label/Label.tsx:60]: I asked Steven to confirm if this was ok on Basecamp ticket - awaiting his response.

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx (hunk 4 above)
Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx:79]: Sure

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    })
+  }
+
+  const initialValues =
+    selectedBlock != null
+      ? {
+          textResponseHint: selectedBlock.hint ?? ''
+        }
+      : null
+  const maxCharacters = 22
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx:64]: Awaiting Steven to confirm on Basecamp ticket

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx (hunk 4 above)
Comment (storyworks) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/TextResponseFields.tsx]: He ended up privately messaging me - but it's updated

image

File: libs/locales/en/apps-journeys-admin.json [hunk 7]
```diff
@@ -13,6 +13,10 @@
   "Journey Report": "Journey Report",
   "Reports": "Reports",
   "Sign In": "Sign In",
+  "One Row": "One Row",
```
Comment (storyworks) [inline libs/locales/en/apps-journeys-admin.json:16]: In the future would be good to use nesting and pluralisation here. I'm not sure if it's possible unless we use keys..

https://www.i18next.com/translation-function/nesting#passing-nesting-to-interpolated
https://www.i18next.com/translation-function/plurals#singular-plural

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+import { JourneyProvider } from '@core/journeys/ui/JourneyProvider'
+import type { TreeBlock } from '@core/journeys/ui/block'
+import {
+  GetJourney_journey as Journey,
+  GetJourney_journey_blocks_TextResponseBlock as TextResponseBlock
+} from '../../../../../../../../../__generated__/GetJourney'
+import { TEXT_RESPONSE_HINT_UPDATE } from './Hint'
+import { Hint } from '.'
+
+jest.mock('react-i18next', () => ({
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.spec.tsx:14]: Mocking this like the other tests in this project - but ideal future state would show the actual translations in tests

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+            <Form>
+              <TextField
+                id="textResponseHint"
+                name="textResponseHint"
+                variant="filled"
+                label="Hint"
+                fullWidth
+                value={values.textResponseHint}
+                inputProps={{ maxLength: maxCharacters }}
+                helperText={t('Can only be {{maxCharacters}} characters', {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx:81]: Made this translation

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/ToggleButtonGroup/ToggleButtonGroup.tsx [hunk 10]
```diff
@@ -24,61 +26,52 @@ export function ToggleButtonGroup({
...
-            <Typography variant="subtitle2">{label}</Typography>
-          ) : (
-            label
-          )}
-        </ToggleButton>
-      ))}
-    </MuiToggleButtonGroup>
+    <Stack sx={{ px: 6, py: 4 }}>
+      {label != null ? (
+        <Typography variant="subtitle2" gutterBottom>
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/ToggleButtonGroup/ToggleButtonGroup.tsx:31]: Added label to ToggleButtonGroup

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Label/Label.tsx [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+        <Formik initialValues={initialValues} onSubmit={noop}>
+          {({ values, errors, handleChange, handleBlur }) => (
+            <Form>
+              <TextField
+                id="textResponseLabel"
+                name="textResponseLabel"
+                variant="filled"
+                label="Label"
+                fullWidth
+                value={
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Label/Label.tsx:76]: I think the value should only reset when the user tries to submit. It's kinda inconvenient never being able to clear the text field

Comment (storyworks) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Label/Label.tsx]: Thanks for finding this!

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+        <Formik initialValues={initialValues} onSubmit={noop}>
+          {({ values, errors, handleChange, handleBlur }) => (
+            <Form>
+              <TextField
+                id="textResponseHint"
+                name="textResponseHint"
+                variant="filled"
+                label="Hint"
+                fullWidth
+                value={values.textResponseHint}
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx:79]: When you save this field when its empty, it shifts the layout for the block. Idk if its major enough to keep us merging tho...

**Review by csiyang — APPROVED** [review]

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx (hunk 12 above)
Comment (storyworks) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/TextResponse/TextResponseFields/Hint/Hint.tsx:79]: Will fix in cooldown

**Pull request merged**